tuner/tuner/libtuner.py

```diff
@@ -598,6 +642,172 @@ def run_command(run_pack: RunPack) -> RunResult:
     return RunResult(result, is_timeout)
 
 
+# The `strip_root_op_attr` and `strip_compilation_info` functions are used for
+# getting consistent inputs to the compilation step in tuning. Inputs may come
+# in with lowering configs, translation info, and root_op attrs when the input
+# is a benchmark, but not when the input is a source MLIR file. Stripping the
+# info makes the inputs to compilation consistent, and allows for overwriting
+# the compilation info with generated TD specs during codegen.
+def strip_root_op_attr(module: ir.Module):
+    root_ops: list[ir.Operation] = get_ops_from_module(module, is_root_op)
+    for root_op in root_ops:
+        assert (
+            ROOT_OP_ATTR_NAME in root_op.opview.attributes
+        ), f"expected root op to have '{ROOT_OP_ATTR_NAME}' attr"
+        del root_op.opview.attributes[ROOT_OP_ATTR_NAME]
+
+
+# See the above comment for `strip_root_op_attr`.
+def strip_compilation_info(input_path: Path) -> str:
+    # Strip compilation info from the source and save the stripped IR
+    strip_command = [
+        f"iree-opt",
+        f"{input_path}",
+        f"--iree-codegen-strip-compilation-info",
+    ]
+    try:
+        result = subprocess.run(
+            strip_command,
+            capture_output=True,
+            text=True,
+        )
+        stripped_mlir = result.stdout
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        cmd = " ".join(strip_command)
+        logging.error(f"Command '{cmd}' returned non-zero exit status {e.returncode}.")
+        logging.error(f"Command '{cmd}' failed with error: {e.stderr}")
```

Comment:
I think it'd be cleaner to define a helper function to launch external processes and have uniform error handling in the tuner

Reply:
I could do this, but it will only be used in this strip_compilation_info function right now. The compilation and benchmarking happens within the python bindings, and they raise their own exception types, so right now this is the only place something like that would be used. I'm fine with adding it anyway, though, if you think it is cleaner that way.

Reply:
I used the existing run_command helper, which has some error handling.